syntax = "proto3";

package vfl;

message DatasetInfo {
  string dataset_name = 1;
  string match_id_name = 2;
  string dtype = 3;
  repeated string features = 4;
  string label_name = 5;
  float numerical_value_of_nan = 6;
}

message UploadDatasetRequest {
  DatasetInfo dataset = 1;
}

message WorkResponse {
  string work_uuid = 1;
}

message StatusRequest {
  string work_uuid = 1;
}

message StatusResponse {
  WorkStatus status = 1;
  string message = 2;
}

message TrainingParams {
  string params = 1;
}

message TrainingRequest {
  TrainingParams params = 1;
  string dataset_name = 2;
  repeated string features = 4;
  string output_model_name = 6;
}

message PredictRequest {
  string model_name = 1;
  string dataset_name = 2;
  repeated string features = 3;
  string output_scores_name = 4;
}

enum WorkStatus {
  WORK_STATUS_IN_PROGRESS = 0;
  WORK_STATUS_COMPLETED = 1;
  WORK_STATUS_FAILED = 2;
  WORK_STATUS_NOT_FOUND = 3;
}
